dotnet-desktop-guide/net/winforms/printing/how-to-print-text-document.md

```diff
@@ -0,0 +1,53 @@
+---
+title: "How to print multi-page text file"
+description: Learn how to print multiple page text file (Windows Forms .NET).
+ms.date: "04/29/2022"
+dev_langs: 
+  - "csharp"
+  - "vb"
+helpviewer_keywords: 
+  - "printing [Windows Forms], printing multiple pages"
+  - "text [Windows Forms], printing Windows Forms"
+  - "Windows Forms, printing text"
+  - "printing [Windows Forms], text"
+ms.custom: devdivchpfy22
+---
+
+# How to print a multi-page text file (Windows Forms .NET)
+
+It's common for Windows-based applications to print text. The <xref:System.Drawing.Graphics> class provides methods for drawing objects (graphics or text) to a device, such as a screen or printer. The following section describes in detail the process to print text file. This method doesn't support printing non-plain text files, such as an Office Word document or a _PDF_ file.
+
+> [!NOTE]
+> The <xref:System.Windows.Forms.TextRenderer.DrawText%2A> methods of <xref:System.Windows.Forms.TextRenderer> are not supported for printing. You should always use the <xref:System.Drawing.Graphics.DrawString%2A> methods of <xref:System.Drawing.Graphics>, as shown in the following code example, to draw text for printing purposes.
+
+## To print text
+
+01. Add a <xref:System.Drawing.Printing.PrintDocument> component.
+
+01. Add a string variable to the class.
+
+    :::code language="csharp" source="snippets/how-to-print-text-document/csharp/Form1.cs" id="add_string_to_your_form":::
+
+    :::code language="vb" source="snippets/how-to-print-text-document/vb/Form1.vb" id="add_string_to_your_form":::
+
+01. To print a document, set the <xref:System.Drawing.Printing.PrintDocument.DocumentName%2A> property to the document you wish to print. Then open and read the document content to the string you added previously. Code to open and read document can be added to the button click event handler. Call the <xref:System.Drawing.Printing.PrintDocument.Print%2A> method to raise the <xref:System.Drawing.Printing.PrintDocument.PrintPage> event. As shown in the highlighted code below, the method `.Print()` will start printing and method raise the `PrintPage` event.
+
+    :::code language="csharp" source="snippets/how-to-print-text-document/csharp/Form1.cs" id="set_DocumentName_and_string" highlight= "10-12":::
+
+    :::code language="vb" source="snippets/how-to-print-text-document/vb/Form1.vb" id="set_DocumentName_and_string" highlight= "17-19":::
```

Comment:
I rewrote this paragraph to be clearer and then fixed the highlighting range to the updated code suggestions
```suggestion
01. Back in the `Click` event handler code, set the <xref:System.Drawing.Printing.PrintDocument.DocumentName%2A> property to the name of the document. This information is sent to the printer. Next, read the document text content and store it in the `stringToPrint` string. Finally, call the <xref:System.Drawing.Printing.PrintDocument.Print%2A> method to raise the <xref:System.Drawing.Printing.PrintDocument.PrintPage> event. The `Print` method is highlighted below.

    :::code language="csharp" source="snippets/how-to-print-text-document/csharp/Form1.cs" id="set_DocumentName_and_string" highlight= "12":::

    :::code language="vb" source="snippets/how-to-print-text-document/vb/Form1.vb" id="set_DocumentName_and_string" highlight= "12":::
```